otx/algorithms/segmentation/adapters/mmseg/models/segmentors/mean_teacher_segmentor.py

```diff
@@ -21,19 +23,42 @@ class MeanTeacherSegmentor(BaseSegmentor):
     It creates two models and ema from one to the other for consistency loss.
     """
 
-    def __init__(self, orig_type=None, unsup_weight=0.1, semisl_start_iter=30, **kwargs):
+    def __init__(
+        self,
+        orig_type=None,
+        unsup_weight=0.1,
+        proto_weight=0.7,
+        aux_weight=0.1,
+        drop_percent=80,
+        num_iters_per_epoch=6000,
+        semisl_start_iter=1,
+        proto_head=None,
+        **kwargs
+    ):
         super().__init__()
         self.test_cfg = kwargs["test_cfg"]
-        self.semisl_start_iter = semisl_start_iter
         self.count_iter = 0
-
+        self.filter_pixels_iters = num_iters_per_epoch * 100  # 100 epochs
+        self.semisl_start_iter = num_iters_per_epoch * semisl_start_iter  # 1 epoch
+        self.drop_percent = drop_percent
+        self.aux_weight = aux_weight
         cfg = kwargs.copy()
         cfg["type"] = orig_type
         self.align_corners = cfg["decode_head"].align_corners
         self.model_s = build_segmentor(cfg)
         self.model_t = build_segmentor(cfg)
+        self.use_prototype_head = False
         self.unsup_weight = unsup_weight
-
+        if proto_head is not None and hasattr(self.model_s.decode_head, "_forward_feature"):
+            self.proto_net = ProtoNet(num_classes=self.model_s.decode_head.num_classes, **proto_head)
+            self.use_prototype_head = True
+        elif proto_head is not None:
+            logger.warning(
+                "Prototype head isn't supported by this model. "
```

Comment:
how about to add comment about what will happen in this case?

Reply:
It looks like if self.use_prototype_head is False, it runs with naive mean teacher method. So I think it would be good to add the message something like 'mean teacher without prototype is used for semi-supervised learning.'